transportable-udfs-avro/src/main/java/com/linkedin/transport/avro/AvroWrapper.java

```diff
@@ -33,29 +29,38 @@
   private AvroWrapper() {
   }
 
-  public static StdData createStdData(Object avroData, Schema avroSchema) {
+  public static Object createStdData(Object avroData, Schema avroSchema) {
     switch (avroSchema.getType()) {
       case INT:
-        return new AvroInteger((Integer) avroData);
       case LONG:
-        return new AvroLong((Long) avroData);
       case BOOLEAN:
-        return new AvroBoolean((Boolean) avroData);
+        return avroData;
       case STRING:
-        return new AvroString((Utf8) avroData);
+        return avroData == null? null : avroData.toString();
       case ARRAY:
-        return new AvroArray((GenericArray<Object>) avroData, avroSchema);
+        return new AvroArrayData((GenericArray<Object>) avroData, avroSchema);
       case MAP:
-        return new AvroMap((Map<Object, Object>) avroData, avroSchema);
+        return new AvroMapData((Map<Object, Object>) avroData, avroSchema);
       case RECORD:
-        return new AvroStruct((GenericRecord) avroData, avroSchema);
+        return new AvroRowData((GenericRecord) avroData, avroSchema);
       case NULL:
         return null;
       default:
         throw new RuntimeException("Unrecognized Avro Schema: " + avroSchema.getClass());
     }
   }
 
+  public static Object getPlatformData(Object transportData) {
+    if (transportData instanceof Integer || transportData instanceof Long || transportData instanceof Boolean) {
+      return transportData;
+    } else if (transportData instanceof String) {
+      return transportData == null? null : new Utf8((String) transportData);
```

Comment:
Check for nulls at the start of the method like in other platforms.

Reply:
It is handled by the primitive type nullability. This way leads to less branchings. In other platforms it is needed as the platform type and the Transport type do not coincide.